lib/srv/db/cloud/iam.go

```diff
@@ -125,3 +192,119 @@ func (c *IAM) getAWSIdentity(ctx context.Context) (aws.Identity, error) {
 	c.awsIdentity = awsIdentity
 	return c.awsIdentity, nil
 }
+
+// getPolicyName returns the inline policy name.
+func (c *IAM) getPolicyName() (string, error) {
+	clusterName, err := c.cfg.AccessPoint.GetClusterName()
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	prefix := clusterName.GetClusterName()
+
+	// If the length of the policy name is over the limit, trim the cluster
+	// name from right and keep the policyNameSuffix intact.
+	maxPrefixLength := maxPolicyNameLength - len(policyNameSuffix)
+	if len(prefix) > maxPrefixLength {
+		prefix = prefix[:maxPrefixLength]
+	}
+
+	return prefix + policyNameSuffix, nil
+}
+
+// processTask runs an IAM task.
+func (c *IAM) processTask(ctx context.Context, task iamTask) error {
+	configurator, err := c.getAWSConfigurator(ctx, task.database)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// TODO(greedy52) ideally tasks can be bundled so the semaphore is acquired
+	// once per group, and the IAM policy is only get/put once per group.
+	lease, err := services.AcquireSemaphoreWithRetry(ctx, services.AcquireSemaphoreWithRetryConfig{
+		Service: c.cfg.AccessPoint,
+		Request: types.AcquireSemaphoreRequest{
+			SemaphoreKind: configurator.cfg.policyName,
+			SemaphoreName: configurator.cfg.identity.GetName(),
+			MaxLeases:     1,
+			Expires:       c.cfg.Clock.Now().Add(time.Minute),
+		},
+		Retry: utils.LinearConfig{
+			Step:   10 * time.Second,
+			Max:    2 * time.Minute,
```

Comment:
Should we make those configurable? If not can you add a comment to explain why we're using these values?